add process documentation (#771)
SovereignCloudStack/website: README.md

# The SCS website project

The static [SCS website](https://scs.community) is generated by [Jekyll](https://jekyllrb.com/). The theme is originally based on the template [business-frontpage](https://github.com/sharu725/business-frontpage) by [sharu725](https://github.com/sharu725). We're heavily making use of [Bootstrap 5](https://getbootstrap.com/docs/5.0) to layout the page content.

The website is automatically build every 15 minutes or on every push to `main` and subsequently pulled onto our webspace.

## Locations

Published sites:

* [The productive SCS site](https://scs.community/)
* [The staging area](https://staging.scs.community/)

## The basic editorial workflow

The website data is deployed by the following process:

![Website workflow](_assets/images/readme/website_workflow.drawio.svg)

The following steps describe the basic workflow how to add, change and remove content.
The description is a little more detailed than it needs to be, as it aims also to provide a reliable guide for non-engineer colleagues.

1. Create a branch based on the `main` branch
   (you can also do this using the github user interface)
   ```

>    git clone git@github.com:SovereignCloudStack/website.git # you only only the fist time

Reviewer: ```suggestion
   git clone git@github.com:SovereignCloudStack/website.git # you only need to do this the first time
```

# The SCS website project

The static [SCS website](https://scs.community) is generated by [Jekyll](https://jekyllrb.com/). The theme is originally based on the template [business-frontpage](https://github.com/sharu725/business-frontpage) by [sharu725](https://github.com/sharu725). We're heavily making use of [Bootstrap 5](https://getbootstrap.com/docs/5.0) to layout the page content.

The website is automatically build every 15 minutes or on every push to `main` and subsequently pulled onto our webspace.

## Locations

Orientation for newbies:

* [Place for adding further employees](https://github.com/SovereignCloudStack/website/tree/main/_employees)
* [Place for adding further community profiles](https://github.com/SovereignCloudStack/website/tree/main/_members)
* [Place for adding further partners](https://github.com/SovereignCloudStack/website/tree/main/_partners)
* [Place for adding further supporting partners](https://github.com/SovereignCloudStack/website/blob/main/_data/supporter.yml)
* [Place for adding further providers](https://github.com/SovereignCloudStack/website/blob/main/_data/provider.yml)

Published sites:

* [The productive SCS site](https://scs.community/de/)

> * [The staging area](https://staging.scs.community/de/)

Author: No, shortened it.

## The basic editorial workflow

The website data is deployed by the following process:

![Website workflow](_assets/images/readme/website_workflow.drawio.svg)

> The following steps describe the basic workflow how to add, change and remove content.

Reviewer 2: ```suggestion
The following steps describe the basic workflow on how to add, change and remove content.
```

> The description is a little more detailed than it needs to be, as it aims also to provide a reliable guide for non-engineer colleagues.

Reviewer 2: While I understand the reasoning, I think this makes reading this documentation cumbersome (and as such leads to people not reading it ;)
The same methods of working (sign-off etc.) are used for all the repos (to which non-engineer colleagues contribute as well). This leads to duplication of documentation (and as a mid-term result to outdated docs). Of course, this is minor commentary and nothing that should block merging this.

Author: IMHO is the process to complicated for non-engineers. The experience of the last weeks was that new changes are typically  performed in collaboration

1. Create a branch based on the `main` branch
   (you can also do this using the github user interface)
   ```
   git clone git@github.com:SovereignCloudStack/website.git # you only only the fist time
   cd website
   git checkout main
   git pull
   git checkout -b feat/<name of your branch> # not not use spaces
   ``` 
2. Make your changes with the tool of your choice
   <details>
   <summary>Visual Studio Code</summary>

>      - Download and install [Visual Studio Code](https://code.visualstudio.com/) Development Environment on your workstation

Reviewer 2: Why do we advocate for VS Code? Any specific reason?

Author: No, clarified that.

     - Install the [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) Extension
       - Preferences -> Extensions
     - Add search extension: "Markdown All in One"
     - Add spellcheck extension "Code Spell Checker"
     - Select the extension and hit the button "Install"
   - Open the `website` folder with Visual Studio Code
   - Open files for editing and use the preview folder
  </details>

1. Commit and push with signoffs
   ```
   git status
   git diff
   git add <file> ... <file>
   git status
   git commit -s -a
   gut push 
   ```

> 2. Optional: Testing complexer changes

Reviewer 2: ```suggestion
2. Optional: Testing more complex changes
```

## Testing changes on `staging` branch

> We're using a dedicated workflow to build a staging preview of our website that subsequently is published with GitHub Pages. To make use of this feature, simply push to `staging`. Please try to keep `staging` in sync with `main`, e.g. by deleting and recreating after testing your recent changes. The staging area can be reached via <https://sovereigncloudstack.github.io/website/>.

Reviewer 2: Shouldn't this be removed / reworded in favour of the addition above?

Author: Yes. Done
